config/routes.js

```diff
@@ -543,6 +543,24 @@ module.exports = [
       validate: {
         params: Joi.object({
           id: Joi.string().regex(objectIdRegex),
+          email: Joi.string().email({ minDomainSegments: 2 }),
```

Comment:
@orakili this is the one absolutely necessary line of the PR. The rest was just some optional cleanup stuff I threw in. Avoiding hoisting and all that..